### PR TITLE
feat(tests): add exdev tests

### PR DESCRIPTION
**Pull request by musikid**

### CODE REVIEW

File: rust/src/tests/errors/exdev.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+
+/// Guard which checks if a secondary file system has been configured.
+pub(crate) fn secondary_fs_available(config: &Config, _: &Path) -> anyhow::Result<()> {
+    config
+        .features
+        .secondary_fs
+        .is_some()
+        .then_some(())
+        .ok_or_else(|| anyhow::anyhow!("No secondary file-system has been configured."))
```
Comment (asomers) [inline rust/src/tests/errors/exdev.rs:17]: simplification:
```suggestion
        .map_or_else(|| Err(anyhow::anyhow!("No secondary file-system has been configured.")), drop)
```

**Review by asomers — CHANGES_REQUESTED** [review]
This looks pretty good.  Is there any reason that it's still in the "draft" state?  Also, we should probably change the CI configuration to get coverage for this.

**Comment by musikid** [comment]
Yes, that's the reason why I let it as a draft, I wasn't sure on how to configure the CI for this, especially with the Docker problems with mounts.

**Review by asomers — APPROVED** [review]
I suggest you merge it as-is, and then I'll add the secondary file system for CI.

**Pull request merged**